website_event_membership_restriction/views/event_templates.xml

```diff
@@ -18,9 +18,11 @@
             expr="//form//div[@id='o_wevent_tickets_collapse']//button"
             position="after"
         >
+            <t t-set="user" t-value="request.env.user" />
+            <t t-set="partner" t-value="user.partner_id" />
             <t
                 t-set="is_disabled_for_non_members"
-                t-value="event.registration_membership_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free'] or request.env.user.partner_id.membership_state not in ['paid', 'free'])"
+                t-value="event.registration_membership_only and (not user or (partner.commercial_partner_id.membership_state not in ['paid', 'free'] and partner.membership_state not in ['paid', 'free']))"
```

Comment:
Can't it be the case that the commercial partner is a partner and the partner is not, and vice versa?
```suggestion
                t-value="event.registration_membership_only and (not user or (partner.commercial_partner_id.membership_state not in ['paid', 'free'] or partner.membership_state not in ['paid', 'free']))"
```
???

Reply:
To disable the button, both the partner and the commercial partner must be non member. What you are suggesting is again the same that it was, which was incorrect.